### PR TITLE
[ETCM-134] Handle failed transactions in history

### PR DESCRIPTION
**Pull request by kapke**

### CODE REVIEW

File: src/common/wallet-state.ts [hunk 1]
```diff
@@ -210,6 +210,30 @@ const getStatus = (tx: AccountTransaction, currentBlock: number): Transaction['s
   else return 'confirmed'
 }
 
+export const mergeTransactions = (current: Transaction[]) => (
+  previous: Transaction[],
+): Transaction[] => {
+  const currentByHashes = new Map(current.map((tx) => [tx.hash, tx]))
```
Comment (conqeror) [inline src/common/wallet-state.ts:216]: 👍

File: src/common/wallet-state.ts [hunk 2]
```diff
@@ -210,6 +210,30 @@ const getStatus = (tx: AccountTransaction, currentBlock: number): Transaction['s
...
+
+  const previousOutput = previous
```
Comment (conqeror) [inline src/common/wallet-state.ts:218]: This should contain only failed transactions, right? Can't we find a better name than `previousOutput` then? :)

File: src/common/wallet-state.test.ts [hunk 3]
```diff
@@ -38,25 +39,67 @@ const pending1 = createTestTx('outgoing', 'pending', 100, 10)
...
+    const failed: Transaction = {...pending1, status: 'failed'}
+    const previous = [...pendingTestBaseTransactions, failed]
+    const current = pendingTestBaseTransactions
+
+    const result = mergeTransactions(current)(previous)
+
+    assert.deepEqual(result, [failed, ...pendingTestBaseTransactions])
+  })
+
+  it('keeps disappearing confirmed and persisted transactions', () => {
```
Comment (conqeror) [inline src/common/wallet-state.test.ts:87]: Shouldn't this be actually 'keeps confirmed and persisted transactions'? Actually, on the fourth read, I understand what it tries to say, but can be understood both ways - (keeps disappearing) (confirmed and persisted transactions). Maybe we can go with 'keeps confirmed and persisted transactions which disappeared'?

File: src/common/wallet-state.ts (hunk 2 above)
Comment (kapke) [reply src/common/wallet-state.ts:218]: Only failed and confirmed, etc. that are not present in the newer list

**Review by conqeror — APPROVED** [review]

**Pull request merged**